src/sentry/api/endpoints/project_details.py

```diff
@@ -142,7 +175,24 @@ def validate(self, data):
         try:
             config_str = json.dumps(data)
             validate_sampling_configuration(config_str)
-            self.validate_uniform_sampling_rule(data.get("rules", []))
+
+            rules = data.get("rules", [])
+            self.validate_uniform_sampling_rule(rules=rules)
+            authorization_result = self.authorize_dynamic_sampling(
+                organization=self.context["project"].organization,
+                user=self.context["request"].user,
+                rules=rules,
+            )
+            # We want to keep different authorization results because in the future we might act differently
+            # based on each specific case.
+            if authorization_result in (
+                DynamicSamplingAuthorizationResult.DS_NOT_AUTHORIZED,
+                DynamicSamplingAuthorizationResult.DS_UNIFORM_NOT_AUTHORIZED,
+                DynamicSamplingAuthorizationResult.DS_CONDITION_NOT_AUTHORIZED,
+            ):
+                raise serializers.ValidationError(
+                    "You do not have permission to set a sampling rule."
+                )
```

Comment:
This logic can be moved to `self.authorize_dynamic_sampling` and then you won't need the Enum class altogether which would simplify the implementation